ui/mirage/handlers/custom-messages.js

```diff
@@ -0,0 +1,120 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: BUSL-1.1
+ */
+
+export default function (server) {
+  server.get('/sys/config/ui/custom-messages', (schema, request) => {
+    if (request.queryParams.authenticated === 'true') {
```

Comment:
I've tested this by trying to fetch from this endpoint in the custom-message route and noticed that queryParams properties get returned as strings.